packages/realm-server/tests/realm-endpoints/lint-test.ts

```diff
@@ -42,130 +54,821 @@ module(`realm-endpoints/${basename(__filename)}`, function () {
+@field name = contains(StringField, { description: "test description" });
+}`);
+
+        assert.strictEqual(response.status, 200, 'HTTP 200 status');
+        let responseJson = JSON.parse(response.text);
+
+        // Should use single quotes based on prettier configuration
+        assert.ok(
+          responseJson.output.includes("'https://cardstack.com/base/string'"),
+          'Single quotes are used for imports',
+        );
+        assert.ok(
+          responseJson.output.includes("'https://cardstack.com/base/card-api'"),
+          'Single quotes are used consistently',
+        );
+        assert.ok(
+          responseJson.output.includes("'test description'"),
+          'Single quotes are used for string literals',
+        );
+      });
+
+      test('supports X-Filename header for parser detection', async function (assert) {
+        let response = await request
+          .post('/_lint')
+          .set(
+            'Authorization',
+            `Bearer ${createJWT(testRealm, 'john', ['read', 'write'])}`,
+          )
+          .set('X-HTTP-Method-Override', 'QUERY')
+          .set('Accept', 'application/json')
+          .set('X-Filename', 'my-card.gts')
+          .send(`import { CardDef } from 'https://cardstack.com/base/card-api';
+export class MyCard extends CardDef {
+@field name = contains(StringField);
+}
+`);
+
+        assert.strictEqual(response.status, 200, 'HTTP 200 status');
+        let responseJson = JSON.parse(response.text);
+        assert.strictEqual(
+          responseJson.output,
+          `import StringField from 'https://cardstack.com/base/string';
+import { CardDef, field, contains } from 'https://cardstack.com/base/card-api';
+export class MyCard extends CardDef {
+  @field name = contains(StringField);
+}
+`,
+          'X-Filename header is used for parser detection',
+        );
+      });
+
+      test('handles large files within reasonable time', async function (assert) {
+        // Create a large file with many imports and classes
+        let largeContent = `import { CardDef } from 'https://cardstack.com/base/card-api';\n`;
+        for (let i = 0; i < 100; i++) {
+          largeContent += `
+export class MyCard${i} extends CardDef {
+@field name${i} = contains(StringField);
+@field email${i} = contains(EmailField);
+}
+`;
+        }
+
+        let startTime = Date.now();
+        let response = await request
+          .post('/_lint')
+          .set(
+            'Authorization',
+            `Bearer ${createJWT(testRealm, 'john', ['read', 'write'])}`,
+          )
+          .set('X-HTTP-Method-Override', 'QUERY')
+          .set('Accept', 'application/json')
+          .send(largeContent);
+
+        let endTime = Date.now();
+        let duration = endTime - startTime;
+
+        assert.strictEqual(response.status, 200, 'HTTP 200 status');
+        assert.ok(
+          duration < 10000,
+          `Large file processing completed in ${duration}ms (should be < 10s)`,
```

Comment:
10s seems like a lot, I know it’s a huge file but 😳

Reply:
I cut it to 5s